### PR TITLE
Feature/initial template

### PR DESCRIPTION
**Pull request by ddp5730**
* Initial Template
* Build EVT-core as a library
* Build Board source as a library
* Template to build arbitrary number of targets
* Auto-generating documentation template

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[submodule "libs/EVT-core"]
+	path = libs/EVT-core
+	url = ../EVT-core.git
+    # Need to set the full url if not cloned into the EVT organization
```
Comment (ddp5730) [inline .gitmodules:4]: This relative path is really nice, but it only works if the template is cloned into the EVT organization.  And only @cbolles has permissions to do that.  I think it's fine, but I want to draw attention to this.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+   3) If the owner is not set to RIT-EVT then you need to update the .gitmodules file URL
+2) Clone the new repo
+   1) `git clone <URL>`
+3) Update the EVT submodule
+   1) `git submodule update --init --recusive && git pull`
+   2) The above command can be used at any point to update the submodule from remote
+4) Import the project into Read the Docs
+   1) Log in to [readthedocs.io](https://readthedocs.org/)
+   2) Navigate to dashboard page
+   3) Import the Repository.  Follow instructions on webpage
```
Comment (ddp5730) [inline README.md]: Do we want the owner of the readthedocs.io to be the team lead, or just whichever member is in charge of the specific project?  I see this being a problem once members start graduating.

File: .gitmodules (hunk 1 above)
Comment (ddp5730) [reply .gitmodules:4]: readthedocs.org actually fails if you set this as the absolute path, so it either needs to be HTTP or relative.

File: README.md (hunk 2 above)
Comment (cbolles) [reply README.md]: Right now there is an "EVT" account handling the read the docs hosting. We can add "maintainers" to projects, so members would be "maintainers" and the projects would be owned by "EVT"

**Review by cbolles — APPROVED** [review]
I think this looks fantastic. Very well done with this, I think this will make board bring up much faster.

**Review by zsteig — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (ddp5730) [reply README.md]: @cbolles Would you mind updating the README accordingly?  I'm not sure how we set up the hosting with the maintainer accounts.

**Pull request merged**